src/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Unix/DirectoryBasedStoreProvider.cs

```diff
@@ -299,21 +299,24 @@ public void Add(ICertificatePal certPal)
                 }
 
 #if DEBUG
-                // Verify that we're creating files with u+rw and o-rw, g-rw.
+                // Verify that we're creating files with u+rw and u-x, o-rwx, g-rwx.
                 const Interop.Sys.Permissions requiredPermissions =
                     Interop.Sys.Permissions.S_IRUSR |
                     Interop.Sys.Permissions.S_IWUSR;
 
                 const Interop.Sys.Permissions forbiddenPermissions =
+                    Interop.Sys.Permissions.S_IXUSR |
```

Comment:
I'm surprised this is passing.  FileStream today uses S_IRWXU when creating files:
https://github.com/dotnet/corefx/blob/master/src/System.IO.FileSystem/src/System/IO/UnixFileStream.cs#L88-L91